src/syntax/syntax.rkt

```diff
@@ -126,9 +31,14 @@
 
   (hash-set! operators name (apply operator (map (curry hash-ref fields) '(itype otype bf ival)))))
 
-(define-syntax-rule (define-operator (name itypes ...) otype [key value] ...)
-  (register-operator! 'name '(itypes ...) 'otype
-                      (list (cons 'key value) ...)))
+(define-syntax define-operator
+  (syntax-rules ()
+    [(define-operator (name itypes ...) otype [key value] ...)
+     (register-operator! 'name '(itypes ...) 'otype
+                         (list (cons 'key value) ...))]
+    [(define-operator (name . itype) otype [key value] ...)
+     (register-operator! 'name 'itype 'otype
+                         (list (cons 'key value) ...))]))
 
```

Comment:
Nice. Much cleaner.

Reply:
Agreed, though I think having variary functions is still a drag on other parts of the code, so longer-term I think we should get rid of them.